example/WORKSPACE

```diff
@@ -29,10 +29,10 @@ http_archive(
 ### RULES_CONDA ###
```

Comment:
It might seem I'm commenting in random places, but it's better to start threads here than in comments in the PR itself 😛 

> Now I would like to make an automated version of these tests, but the only idea I have for that is to make a bunch of directories under tests/ with their own WORKSPACE and BUILD files and change the use_mamba and clean flags to the matrix of combinations I specified above. Then there would be a test script that goes to each directory and runs bazel test //.... The problem is that that approach would duplicate a lot of files, just to change two flags. Do you have any better idea for structuring the tests?

I was thinking about it too. And for now I don't really have any better idea than duplicating the files 😕

Reply:
Might be worth to take a look how other bazel rules like `rules_python` or `rules_go` tackle that problem

Reply:
I looked at `rules_apple` and it looks like they use a shell script to generate `BUILD` files. We can do something similar for `WORKSPACE` files.

Reply:
Another option is to consider `--experimental_resolved_file_instead_of_workspace` ([design doc](https://blog.bazel.build/2018/09/28/first-class-resolved-file.html)). Although I'm reluctant to use anything marked `--experimental`.

Reply:
> It might seem I'm commenting in random places, but it's better to start threads here than in comments in the PR itself 😛

You prefer to start new threads in the Issue instead of PR? I can do that from now on, sorry.

> Another option is to consider --experimental_resolved_file_instead_of_workspace (design doc). Although I'm reluctant to use anything marked --experimental.

There is also an open bug with that option: https://github.com/bazelbuild/bazel/issues/10933 so I would vote to avoid using it.

> I looked at rules_apple and it looks like they use a shell script to generate BUILD files. We can do something similar for WORKSPACE files.

`rules_python` does that too, for WORKSPACE files. But they use a python file that is run by py_test:

https://github.com/bazelbuild/rules_python/blob/929d5a13d4eb1b930086d9353fc6f2d6ad306e43/tools/bazel_integration_test/bazel_integration_test.bzl
https://github.com/bazelbuild/rules_python/blob/98ffe0655967481026bf112ee172630548b703c2/tools/bazel_integration_test/test_runner.py

I prefer this approach, because it should work on Windows (shell scripts are tricky to get working on Windows) and because it is mostly Bazel-native. But I prefer to implement such a test setup in a separate issue & PR because:

1. Implementing a test runner for integration tests is high effort and deserves a separate code review
2. There will be more time to discuss the tradeoffs in a separate issue
3. `rules_apple` and `rules_python` are licensed `Apache-2.0` so we would likely need to relicense `rules_conda` if we want to borrow any of their code verbatim

Therefore my proposal for now is to implement the tests for this option with just normal directories and BUILD files and try to keep them as small as possible. I will also try to give each test its own WORKSPACE but I'm not sure yet if nested workspaces will work.

Reply:
> You prefer to start new threads in the Issue instead of PR? I can do that from now on, sorry.

No no, I just meant that you can do two kind of comments on PR: "top-level" ones and ones attached to some lines of code. And you can only reply directly to the latter and it starts comment threads (that can be resolved as a bonus), so I always prefer to attach comments to any code, even if they are not related to that particular line 😛 

> But I prefer to implement such a test setup in a separate issue & PR

> Therefore my proposal for now is to implement the tests for this option with just normal directories and BUILD files and try to keep them as small as possible. I will also try to give each test its own WORKSPACE but I'm not sure yet if nested workspaces will work.

Agree, I will create another issue with designing and setting up proper test framework. For now let's stick with duplicated files. As tests will differ in arguments for rules they need different `WORKSPACE` files. It shouldn't be a problem, only `.github/workflows/test.yml` would need to change a little bit. Not sure how to do it the best way, but just copying and pasting the `Run tests` step will be sufficient for now.